docs/architecture/deep-dives/autofill/autofill-menu.md

```diff
@@ -214,86 +215,89 @@ design and implementation of the feature.
+  the injected UI elements.
+
+#### **Reacting to attempts to modify the Auto-fill Menu elements programmatically**
+
+- A defensive approach using the
+  [MutationObserver API](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) is
+  implemented to guard against modifications to the Auto-fill Menu UI elements.
+- The content script monitors for elements trying to append to the bottom of the body, repositioning
+  the UI elements as necessary to prevent allowing the Auto-fill Menu to be overlaid by potential
+  clickjacking elements.
+- It also watches for changes to the styles or attributes of the Auto-fill Menu UI elements,
+  resetting them to their original values if altered by malicious actions.
 
 ### Cross-Site Scripting (XSS)
 
 To mitigate the risks of Cross-Site Scripting (XSS), the following defensive measures have been
 implemented for the Auto-fill Menu feature:
 
-- **Limiting User Input Usage in the Auto-fill Menu**
+#### **Limiting user input usage in the Auto-fill Menu**
 
-  - The implementation largely avoids reliance on the injection of external values.
-  - User-created cipher values are used to populate the Auto-fill Menu list with credentials. These
-    values are carefully sanitized when saved in a user's vault.
-  - Collection of foreign input occurs when a user creates a new cipher through the Auto-fill Menu.
-    However, these inputs are processed by the extension background script and are never introduced
-    into any executable context.
+- The implementation largely avoids reliance on the injection of external values.
+- User-created cipher values are used to populate the Auto-fill Menu list with credentials. These
+  values are carefully sanitized when saved in a user's vault.
+- Collection of foreign input occurs when a user creates a new cipher through the Auto-fill Menu.
+  However, these inputs are processed by the extension background script and are never introduced
+  into any executable context.
 
-- **Enforcing a Strict Content Security Policy to Block Foreign Scripts**
-  - The execution of inline scripts or scripts external to the extension is prohibited within the
-    Auto-fill Menu pages. This is achieved by implementing a stringent content security policy
-    specifically for the Auto-fill Menu pages.
+#### **Enforcing a strict content security policy to block foreign scripts**
+
+- The execution of inline scripts or scripts external to the extension is prohibited within the
+  Auto-fill Menu pages. This is achieved by implementing a stringent content security policy
+  specifically for the Auto-fill Menu pages.
 
 ### DOM Clobbering
 
 To address the risks associated with DOM clobbering, the following defensive strategies have been
 implemented for the Auto-fill Menu feature:
 
-- **Utilizing the Isolated Execution Context of the Content Script**
+#### **Utilizing the isolated execution context of the content script.**
```

Comment:
⛏️ Drop the periods at this and the one below.

Reply:
Nice catch, not sure why I added the period. This has been fixed.